build.gradle

```diff
@@ -116,6 +116,38 @@ subprojects {
             force "jdepend:jdepend:${project.jdepend.toolVersion}"
         }
     }
+
+    apply plugin: 'checkstyle'
+    configurations {
+        checkstyleConfig
+    }
+    def versions = [
+            checkstyle: '6.17',
+    ]
+    dependencies {
+        checkstyleConfig ("com.puppycrawl.tools:checkstyle:${versions.checkstyle}") {
+            transitive = false
+        }
+        checkstyle "com.puppycrawl.tools:checkstyle:${versions.checkstyle}"
+    }
+    checkstyle {
+        config = resources.text.fromArchiveEntry(configurations.checkstyleConfig, 'google_checks.xml')
+        ignoreFailures false
+        showViolations false
+    }
+
+    tasks.withType(Checkstyle).each { checkstyleTask ->
+        checkstyleTask.doLast {
+            reports.all { report ->
+                if(report.name == "xml") {
+                    def outputFile = report.destination
+                    if (outputFile.exists() && outputFile.text.contains("<error ")) {
+                        throw new GradleException("There were checkstyle warnings! For more info check $outputFile")
+                    }
```

Comment:
fine with me.

On Fri, Apr 29, 2016 at 12:49 PM, Etan Shaul <notifications@github.com>
wrote:

> In build.gradle
> <https://github.com/GoogleCloudPlatform/gcloud-intellij/pull/659#discussion_r61607753>
> :
>
> > +        checkstyle "com.puppycrawl.tools:checkstyle:${versions.checkstyle}"
> > +    }
> > +    checkstyle {
> > +        config = resources.text.fromArchiveEntry(configurations.checkstyleConfig, 'google_checks.xml')
> > +        ignoreFailures false
> > +        showViolations false
> > +    }
> > +
> > +    tasks.withType(Checkstyle).each { checkstyleTask ->
> > +        checkstyleTask.doLast {
> > +            reports.all { report ->
> > +                if(report.name == "xml") {
> > +                    def outputFile = report.destination
> > +                    if (outputFile.exists() && outputFile.text.contains("<error ")) {
> > +                        throw new GradleException("There were checkstyle warnings! For more info check $outputFile")
> > +                    }
>
> I discussed with with @meltsufin <https://github.com/meltsufin> and
> @chanseokoh <https://github.com/chanseokoh> yesterday and I think the
> best option is to fail the build locally. This way any of those
> verification problems are discovered early.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/gcloud-intellij/pull/659/files/5bd5e00b0c4b8ad6735f377c77bbdebb525d22d1#r61607753>
>